subnet/watch.go

```diff
@@ -77,9 +77,13 @@ func (lw *leaseWatcher) reset(leases []Lease) []Event {
 
 		found := false
 		for i, ol := range lw.leases {
-			if ol.Subnet.Equal(nl.Subnet) && bytes.Compare(ol.Attrs.BackendData, nl.Attrs.BackendData) == 0 {
+			if ol.Subnet.Equal(nl.Subnet) {
 				lw.leases = deleteLease(lw.leases, i)
-				found = true
+
+				// If the backend data has changed, send the added event to the backend
+				if bytes.Compare(ol.Attrs.BackendData, nl.Attrs.BackendData) == 0 {
+					found = true
```

Comment:
Will it do an "upsert" upon getting the "added" event? I.e. we don't need to issue a "removed" event first, do we?

Reply:
No we don't need to send a remove event. Tracing a backend data change under normal circumstances only seems to produce an add event, and the only events defined are added and removed: https://github.com/gravitational/flannel/blob/master/subnet/subnet.go#L63-L66